lib/api/api_auth.go

```diff
@@ -52,21 +60,37 @@ func hasAnyPrefix(s string, prefixes []string) bool {
 	return false
 }
 
+func noAuthPaths() []string {
+	return []string{
+		"/",
+		"/index.html",
+		"/modal.html",
+	}
+}
+
 func noAuthPrefixes() []string {
 	return []string{
+		// Static assets
+		"/assets/",
+		"/syncthing/",
+		"/vendor/",
+		"/theme-assets/", // This leaks information from config, but probably not sensitive
+
+		// No-auth API endpoints
 		"/rest/noauth",
 	}
 }
 
-func basicAuthAndSessionMiddleware(cookieName string, guiCfg config.GUIConfiguration, ldapCfg config.LDAPConfiguration, next http.Handler, evLogger events.Logger) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func authAndSessionMiddleware(cookieName string, guiCfg config.GUIConfiguration, ldapCfg config.LDAPConfiguration, next http.Handler, evLogger events.Logger) (http.Handler, http.Handler) {
+
+	handleAuthzPassthrough := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
What's the "z" stand for in "Authz"?

Reply:
"Authori**z**ation" (permissions management), to differentiate from "authenti**c**ation" (integrity and identity verification).

But it's actually a misnomer... in the commit that introduced this, there was just HTTP Basic Authorization under the hood of this, so the name comes from that. But the "authorization" in that is also a misnomer because it's really dealing with authentication, and `handleAuthzPassthrough` is even more so now that there's also WebAuthn under the hood.

Reply:
Fixed along with https://github.com/syncthing/syncthing/pull/8757#discussion_r1089746400.